NoticeboardAPI + update with Review plan API (due to wrong branch name)

thuthichchoichu opened the pull request:
Result functions: removed suspend, added inline instead;

ionate12 commented on app/src/main/java/au/com/safetychampion/MainViewModel.kt:
```diff
@@ -341,12 +349,12 @@ class MainViewModel : ViewModel() {
     }
 
-    private val fetchListDocUsecase: FetchDocumentUseCase by koinInject()
+    private val fetchListDocUsecase: FetchListDocumentUseCase by koinInject()
```
Fetch -> Get

ionate12 commented on app/src/main/java/au/com/safetychampion/MainViewModel.kt:
```diff
@@ -341,12 +349,12 @@ class MainViewModel : ViewModel() {
     }
 
-    private val prepareDocUseCase: FetchDocumentUseCase by koinInject()
+    private val prepareDocUseCase: PrepareDocumentSignoffUseCase by koinInject()
```
PrepareDocumentSignoff -->  PrepareSignoffDocument

khoimaianest committed:
NoticeboardAPI.
Deleted NoticeboardCell(s)

ionate12 commented on data/src/main/java/au/com/safetychampion/data/data/incident/IIncidentRepository.kt:
```diff
@@ -0,0 +1,19 @@
+interface IIncidentRepository {
+    suspend fun list(): Result<List<Incident>>
+    suspend fun new(body: IncidentNewPL): Result<Incident>
+    suspend fun fetchIncident(taskID: String): Result<Incident>
```
1.đổi hết: ID --> id 
2.fetchIncident -> fetch
3.ở đây ko phải là taskId mà là incidentId(moduleId)

ionate12 commented on data/src/main/java/au/com/safetychampion/data/data/incident/IIncidentRepository.kt:
```diff
@@ -0,0 +1,19 @@
+    suspend fun list(): Result<List<Incident>>
+    suspend fun new(body: IncidentNewPL): Result<Incident>
+    suspend fun fetchIncident(taskID: String): Result<Incident>
+    suspend fun fetchTask(taskID: String): Result<IncidentTask>
```
taskId -> incidentId

ionate12 commented on data/src/main/java/au/com/safetychampion/data/data/incident/IncidentRepositoryImpl.kt:
```diff
@@ -0,0 +1,77 @@
+import kotlinx.coroutines.async
+import kotlinx.coroutines.withContext
+
+class IncidentRepositoryImpl : BaseRepository(), IIncidentRepository {
```
e sửa lại hết các class RepoImpl --> Repo thôi. vs `IncidentRepositoryImpl` -> `IncidentRepository`

ionate12 commented on data/src/main/java/au/com/safetychampion/data/data/noticeboard/NoticeboardBlockList.kt:
```diff
@@ -0,0 +1,21 @@
+import au.com.safetychampion.data.domain.uncategory.DocAttachment
+
+data class NoticeboardBlockList(
+    val attachments: MutableList<DocAttachment>? = null,
```
DocAttachment -> Attachent

nên dùng IAttachment để dễ quản lý

ionate12 commented on data/src/main/java/au/com/safetychampion/data/domain/models/ModelComponents.kt:
```diff
@@ -27,12 +27,16 @@ interface ISubcategoryCusval : IBaseCusval {
 
 interface ISignature {
-    var signatures: MutableList<Signature>
+    var signatures: MutableList<Signature>?
```
ko có optional nha. default sẽ là listof

ionate12 commented on data/src/main/java/au/com/safetychampion/data/domain/models/crisk/CriskTask.kt:
```diff
@@ -40,7 +40,7 @@ data class CriskTask(
     override var subcategoryCusvals: MutableList<CustomValue>,
-    override var signatures: MutableList<Signature>,
+    override var signatures: MutableList<Signature>?,
     override var attachments: MutableList<Attachment>?,
```
attachments cũng thế ko có `optional`

ionate12 commented on data/src/main/java/au/com/safetychampion/data/domain/models/incidents/BaseIncident.kt:
```diff
@@ -0,0 +1,40 @@
+import au.com.safetychampion.data.domain.models.customvalues.CustomValue
+
+/** Sharing properties between [IncidentNewPL] and [Incident]*/
+sealed interface BaseIncident {
```
implements ICusval, ICategoryCusval, IEnvCusval
cả 2 đều có signature: implement ISignature vào.

ionate12 commented on data/src/main/java/au/com/safetychampion/data/domain/base/BaseSignOff.kt:
```diff
@@ -23,5 +23,7 @@ interface BaseSignOff<T : BaseTask> {
             taskId: String
         ) = "vdocs/$moduleId/tasks/$taskId/signoff"
+
+        fun incidentSignoffSyncableKey(taskId: String) = "incidents/$taskId/task/signoff"
```
cái này là incidentId nha

ionate12 commented on data/src/main/java/au/com/safetychampion/data/domain/models/incidents/IncidentNewPL.kt:
```diff
@@ -0,0 +1,48 @@
+    override var pendingActions: MutableList<PendingActionPL> = mutableListOf() // does not exist in new incident
+) : BasePL(), BaseIncident, IIncidentComponent
+
+interface IIncidentComponent : ICusval, ICategoryCusval, IEnvCusval, ISignature, IAttachment, IPendingAction
```
IPendingAction --> IPendingActionPL

ionate12 commented on data/src/main/java/au/com/safetychampion/data/domain/models/incidents/IncidentNewPL.kt:
```diff
@@ -0,0 +1,48 @@
+    override var propOrEnvDamageDescription: String? = null,
+    var sessionId: String? = null,
+    override var signatures: MutableList<Signature>? = null,
+    override var timeOccurred: TimeField? = null,
```
chưa thấy property của List<ActionLink> ở đây

ionate12 commented on data/src/main/java/au/com/safetychampion/data/domain/models/incidents/IncidentSignOff.kt:
```diff
@@ -1,6 +1,13 @@
+    var body: Incident = Incident(),
+    override var task: IncidentTask = IncidentTask()
+) : BaseSignOff<IncidentTask> {
+    override fun syncableKey() = "incidents/${task._id}/task/signoff"
```
BaseSignoff.getIncidentSyncableKey

khoimaianest committed:
Fixed comment

khoimaianest committed:
Fixed comments:
attachments, signature not null.
BaseIncident impl: ICusval, ICategoryCusval, IEnvCusval, ISignature.
taskID -> incidentId

khoimaianest committed:
RepositoryImpl -> Repository

khoimaianest committed:
updated DocAttachment -> Attachment
